src/main/java/org/opensearch/knn/plugin/rest/RestSearchModelHandler.java

```diff
@@ -69,9 +70,22 @@ private void checkUnSupportedParamsExists(RestRequest request) {
         throw new IllegalArgumentException(errorMessage);
     }
 
+    private void searchSizeValidation(RestRequest request) {
+        ValidationException validationException = new ValidationException();
+        if (request.hasParam("size")) {
```

Comment:
no nested if please. Revert the if to short circuit.

Reply:
We have a line of code in line 76. Adding a short circuit will make the code messy.

Reply:
I think the perforce of both variants we can count as same, it's more for code readability/maintainability. Having multiple smaller methods with well defined scope and name is to me better than one long handyman style method.

Reply:
This isn't messy. But, does short-circuit means adding both conditions in the same if statement?

Reply:
Short-circuit means return as soon as possible. Instead of writing lengthy nested if, change it to if not so that it can exit/return immediately than at end of the method.

Reply:
Got it. Thanks!